baby_bingo.py

```diff
@@ -0,0 +1,60 @@
+
+########## SUPPORTING PACKAGES ##########
+import random
+import re
+import sys
+
+
+########## SUPPORTING FUNCTIONS ##########
+
+def remove_number_prefix(my_string):
+    m = re.match('^[0-9]*[.]\s(.*)', my_string)
+    return(m.group(1))
+
+def remove_rating_suffix(my_string):
+    m = re.match('([a-zA-Z]*)\s(No change)*[+-]*[0-9]*', my_string)
+    return m.group(1)
+
+
+########## CODE TO BE EXECUTED ##########
+
+baby_names = [] # array that will store the baby names
+bingo_card_dim = 4 # the number of rows and columns the bingo card will include
+num_cards_needed = 15 # the number of bingo cards needed
+
+# open, read in, and close the file
+in_file = open('baby_names.txt', 'r')
+
+for line in in_file:
+
+    # clear away the extraneous prefix / suffix strings
+    current_line = remove_rating_suffix(remove_number_prefix(str(line)))
+
+    # store it in a list
+    baby_names.append(current_line)
+
+in_file.close()
+
+
+# open a file to write to, loop through and print the number of needed bingo cards, and close the file
+
+sys.stdout = open('bingo_cards.csv', 'w')

```

Comment:
rather than opening a file and leaving the file handle open the whole time you loop over creating the names, you could build up a single string in the the loop and then once it's finished open the file, write the string and close the file.
I think this would be safer as I don't know what would happen if the loop threw an exception and the file handle was still open.